code/game/objects/structures/spawner.dm

```diff
@@ -40,11 +40,11 @@ GLOBAL_LIST_INIT(astroloot, list(
 	var/mob_types = list(/mob/living/simple_animal/hostile/carp)
 	var/spawn_text = "emerges from"
 	var/faction = list("hostile")
-	var/spawner_type = /datum/component/spawner
+	var/datum/component/spawner_type = /datum/component/spawner // VOID EDIT, allows us to run RemoveComponent on it later
```

Comment:
i don't see the point in this, you shouldn't be calling `RemoveComponent` components remove themselves when you qdel the object

Reply:
I'm not actually using remove component so I'll edit that comment, I am however qdel'ing that component as I want to leave the structure after it has been fully looted. The structure gets looted and the mob spawning component is removed so no more hostiles are created.